sklearn/tests/test_calibration.py

```diff
@@ -13,17 +14,210 @@
+    )
+
+    y_pred_fbeta = clf_fbeta.predict(X_test[calibration_samples:])
+    assert_greater(recall_score(y_test[calibration_samples:], y_pred_fbeta),
+                   recall_score(y_test[calibration_samples:], y_pred))
+
+    clf_max_tpr = CutoffClassifier(
+        lr, strategy='max_tpr', threshold=0.7, cv='prefit'
+    ).fit(X_test[:calibration_samples], y_test[:calibration_samples])
+
+    y_pred_max_tpr = clf_max_tpr.predict(X_test[calibration_samples:])
+
+    tn_max_tpr, fp_max_tpr, fn_max_tpr, tp_max_tpr = confusion_matrix(
+        y_test[calibration_samples:], y_pred_max_tpr).ravel()
+
+    tpr_max_tpr = tp_max_tpr / (tp_max_tpr + fn_max_tpr)
+    tnr_max_tpr = tn_max_tpr / (tn_max_tpr + fp_max_tpr)
+
+    # check that the tpr increases with tnr >= min_val_tnr
+    assert_greater(tpr_max_tpr, tpr)
+    assert_greater(tpr_max_tpr, tpr_roc)
+    assert_greater_equal(tnr_max_tpr, 0.7)
+
+    clf_max_tnr = CutoffClassifier(
+        lr, strategy='max_tnr', threshold=0.7, cv='prefit'
+    ).fit(X_test[:calibration_samples], y_test[:calibration_samples])
+
+    y_pred_clf = clf_max_tnr.predict(X_test[calibration_samples:])
+
+    tn_clf, fp_clf, fn_clf, tp_clf = confusion_matrix(
+        y_test[calibration_samples:], y_pred_clf).ravel()
+
+    tnr_clf_max_tnr = tn_clf / (tn_clf + fp_clf)
+    tpr_clf_max_tnr = tp_clf / (tp_clf + fn_clf)
+
+    # check that the tnr increases with tpr >= min_val_tpr
+    assert_greater(tnr_clf_max_tnr, tnr)
+    assert_greater(tnr_clf_max_tnr, tnr_roc)
+    assert_greater_equal(tpr_clf_max_tnr, 0.7)
+
+    # check error cases
+    clf_bad_base_estimator = CutoffClassifier([])
+    assert_raises(TypeError, clf_bad_base_estimator.fit, X_train, y_train)
+
+    X_non_binary, y_non_binary = make_classification(
+        n_samples=20, n_features=6, random_state=42, n_classes=4,
+        n_informative=4
+    )
+    assert_raises(ValueError, clf_roc.fit, X_non_binary, y_non_binary)
+
+    clf_foo = CutoffClassifier(lr, strategy='f_beta', beta='foo')
+    assert_raises(ValueError, clf_foo.fit, X_train, y_train)
+
+    clf_foo = CutoffClassifier(lr, strategy='foo')
+    assert_raises(ValueError, clf_foo.fit, X_train, y_train)
+
+    for method in ['max_tpr', 'max_tnr']:
+        clf_missing_info = CutoffClassifier(lr, strategy=method)
+        assert_raises(ValueError, clf_missing_info.fit, X_train, y_train)
+
+
+def test_cutoff_cv():
+    X, y = make_classification(n_samples=1000, n_features=6, random_state=42,
+                               n_classes=2)
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y,
+                                                        train_size=0.6,
+                                                        random_state=42)
+    lr = LogisticRegression().fit(X_train, y_train)
+    clf_roc = CutoffClassifier(LogisticRegression(), strategy='roc', cv=3).fit(
+        X_train, y_train
+    )
+
+    y_pred = lr.predict(X_test)
+    y_pred_roc = clf_roc.predict(X_test)
```

Comment:
Please check the threshold and std attributes are sensibly set

Reply:
the only sensible check we can do is to make sure they are set and not 0 I think. did that. let me know what you thnk,